roscala/src/main/scala/coop/rchain/roscala/Vm.scala

```diff
@@ -619,11 +635,14 @@ object Vm {
 
   def getNextStrand(state: State): Boolean =
     if (state.strandPool.isEmpty) {
-      logger.debug("Empty strandPool - exiting VM")
+      logger.debug("Empty strandPool")
       true
     } else {
-      val ctxt = state.strandPool.remove(state.strandPool.size - 1)
-      logger.debug(s"Get next ctxt ($ctxt) and install it")
+      logger.debug(s"Waiting for a Ctxt to complete")
```

Comment:
Are we really waiting here? `strandPool` is not empty and when we get here a `Ctxt` already gave up control.

Reply:
There is a possibility when `ParallelStrandPool` will `join` a task on `deque`, and `join` operation is blocking.